### PR TITLE
Quo Avatars/Token-Avatar Component

### LINKED ISSUE #20089: Buttons/Swap-Order-Button
https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=38822-46877&m=dev

### LINKED ISSUE #20128: Avatars/Token-Avatar
Design: https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=34075-53402&m=dev

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #20128

### Summary

Implements the [Avatars/Token-Avatar ](https://www.figma.com/design/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=34075-53402&m=dev)component

### Testing notes
Manual QA can be skipped since this PR only adds a quo component and doesn't introduce any changes outside Quo Preview.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
None

### Steps to test
Quo Preview -> Avatars -> Token Avatar

status: ready

### CODE REVIEW

File: src/quo/components/avatars/token_avatar/view.cljs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            [react-native.hole-view :as hole-view]
+            [schema.core :as schema]))
+
+(defn view-internal
+  [{:keys [type context? image network-image container-style]}]
+  [rn/view
+   {:style               (merge style/container container-style)
+    :accessibility-label :token-avatar}
+   [hole-view/hole-view
+    {:key   (hash context?)
```
Comment (ajayesivan) [inline src/quo/components/avatars/token_avatar/view.cljs]: The hole is rendered dynamically based on the `context?` prop, and HoleView has an issue in Android(more info here: https://github.com/status-im/status-mobile/issues/15577) using the `key` prop for forceful re-render.

Issue screen recording:

https://github.com/status-im/status-mobile/assets/19279756/1649e6e6-5e22-47e4-8c0f-fb9e915be495

Comment (briansztamfater) [reply src/quo/components/avatars/token_avatar/view.cljs]: If I understood correctly, it in an Android-specific workaround, maybe we can add a `platform/android?` check here.

**Review by briansztamfater — APPROVED** [review]
LGTM!

**Review by mohsen-ghafouri — APPROVED** [review]

File: src/quo/components/avatars/token_avatar/style.cljs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+(ns quo.components.avatars.token-avatar.style)
+
+(def size 32)
```
Comment (Parveshdhull) [inline src/quo/components/avatars/token_avatar/style.cljs]: `(def ^:const size 32)`?

File: src/quo/components/avatars/token_avatar/style.cljs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+   :width  size})
+
+(def hole-view
+  {:width  size
+   :height size})
+
+(def context
+  {:width         16
+   :height        16
+   :border-radius 8
```
Comment (Parveshdhull) [inline src/quo/components/avatars/token_avatar/style.cljs:16]: These also looks like size dependent. (32/2), maybe we can also define these too at the top of file?

File: src/quo/components/avatars/token_avatar/view.cljs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     [:map {:closed true}
+      [:type {:optional true} [:enum :asset :collectible]]
+      [:context? {:optional true} [:maybe :boolean]]
+      [:image :schema.common/image-source]
+      [:network-image {:optional true} [:maybe :schema.common/image-source]]
+      [:container-style {:optional true} [:maybe :map]]]]]
+   :any])
+
+(defn- view-internal
+  [{:keys [type context? image network-image container-style]}]
```
Comment (Parveshdhull) [inline src/quo/components/avatars/token_avatar/view.cljs:21]: Hi, what is context? here and why its boolean?

**Review by Parveshdhull — COMMENTED** [review]
hi @ajayesivan, thank you for your great work.
Sorry for late review. Left some minor comments.

**Review by Parveshdhull — APPROVED** [review]

File: src/quo/components/avatars/token_avatar/view.cljs (hunk 4 above)
Comment (ajayesivan) [reply src/quo/components/avatars/token_avatar/view.cljs:21]: Screenshot 2024-05-24 at 14 35 25
`context?` is the property name defined in Figma that decides whether to display the network image or not.

**Comment by ajayesivan** [comment]
Hi @Francesca-G, could you please do a design review for this PR?

Thanks!

**Review by Francesca-G — APPROVED** [review]
Looks good 👍

**Pull request merged**